unitTests/ui/QueryBuilderTest.ts

```diff
@@ -174,5 +176,39 @@ export function QueryBuilderTest() {
     it('should be able to detect if no end user keywords exist in either the basic or long expression', () => {
       expect(queryBuilder.containsEndUserKeywords()).toBeFalsy();
     });
+
+    it('should have facets undefined if none have been added', () => {
+      expect(queryBuilder.build().facets).toBeUndefined();
+    });
+
+    it('should have facets undefined if none have been added', () => {
+      expect(queryBuilder.build().facets).toBeUndefined();
+    });
+
+    describe('when adding a facet request', () => {
+      const facetRequest: IFacetRequest = { field: '@test' };
+
+      beforeEach(() => {
+        queryBuilder.facetRequests.push(facetRequest);
+      });
+
+      it('should add the facets parameter correctly', () => {
+        expect(queryBuilder.build().facets).toEqual([facetRequest]);
+      });
+
+      it(`if groupByRequests is empty
```

Comment:
Since adding the groupBy request to the queryBuilder no longer depends on the facet request array (and vice-versa), this test and the one on line 205 can be moved out of the `describe` to better reflect the behaviour.